beacon_node/network/src/sync/block_sidecar_coupling.rs

```diff
@@ -55,7 +56,18 @@ impl<T: EthSpec> BlocksAndBlobsRequestInfo<T> {
             if blob_list.is_empty() {
                 responses.push(BlockWrapper::Block(block))
             } else {
-                responses.push(BlockWrapper::BlockAndBlobs(block, blob_list))
+                let mut blobs_fixed = Vec::with_capacity(T::max_blobs_per_block());
+                for blob in blob_list {
+                    let blob_index = blob.index as usize;
+                    if blob_index >= T::max_blobs_per_block() {
+                        return Err("Invalid blob index");
+                    }
+                    blobs_fixed.insert(blob_index, Some(blob));
+                }
+                responses.push(BlockWrapper::BlockAndBlobs(
+                    block,
+                    FixedVector::from(blobs_fixed),
+                ))
```

Comment:
Yes you're right! The intent was to pre-populate with `None`s and update at each blob at the index rather than shifting all the blobs.  What crappy code!

Reply:
Yes you're right! The intent was to pre-populate with `None`s and update at each blob at the index rather than shifting all the blobs.  What crappy code!

Reply:
fixed here: https://github.com/sigp/lighthouse/pull/4152/commits/8c5c92be06bda815612c4041a15c82e92a19f79a